### DISCUSSION TITLE
Faster way for aliases of words?

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#52] [Q&A] [answered]
At the moment I have these:

```
: fconstant 2constant ; 
: f+ d+ ;
: f- d- ;
: fvariable 2variable ;
: fdup 2dup ;
: fswap 2swap ;
: f! 2! ;
: f@ 2@ ;

: di enable-int ;
: ei disable-int ;

```
Is there way to make these so they execute without overhead?

**Comment by tabemann** [comment, marked as answer]
For words that can themselves be inlined you can do, say,

```
: f+ [inlined] d+ ;
: f- [inlined] d- ;
: fdup [inlined] 2dup ;
: fswap [inlined] 2swap ;
: f! [inlined] 2! ;
: f@ [inlined] 2@ ;
: ei [inlined] enable-int ;
: di [inlined] disable-int ;
```

(I assume you accidentally swapped `ei` and `di`.)

Note that this cannot be done with words that are not inlinable, because calls to non-inlined words are inherently non-inlinable due to calls either PC-relative or calls require loading an address from a constant pool beforehand (which itself is PC-relative). In many cases zeptoforth is smart enough to ignore [inlined] if a word cannot be inlined, it should be noted, but if your code crashes instead, please let me know.

**Reply by robzed** [reply]
Good catch on ei/di swap
